File: Algorithms/JavaScript/insertionSort.js
/*
    * insertionSort - to sort a given array
    * @param {array} arr - Array to sort
*/
function insertionSort(arr) {
    for (let i = 0; i < arr.length; i++) {
        let lowest = i
        for (let j = i + 1; j < arr.length; j++) {
            if (arr[lowest] > arr[j]) {
                lowest = j
            }
        }
        if (lowest !== i) {
            let temp = arr[i]
            arr[i] = arr[lowest]
            arr[lowest] = temp   
        }
    }
    return arr
}

console.log(insertionSort([43,56,33,1,22,67,45]))

File: Algorithms/Python/insertionSort.py
# Python program for implementation of Insertion Sort


def insertionSort(arr):
    for i in range(1, len(arr)):
        key = arr[i]
        j = i-1
        while j >= 0 and key < arr[j]:
            arr[j+1] = arr[j]
            j -= 1
        arr[j+1] = key
    return arr


if __name__ == "__main__":
    n = int(input("Enter the no of elements:"))
    ar = []
    print("Enter elements:-")
    for i in range(n):
        ar.append(int(input()))
    ar = insertionSort(ar)
    for j in ar:
        print(j, end=' ')
